### DISCUSSION TITLE
Can't acces v3.1.5 data

### DISCUSSION THREAD

**Discussion by Maxigaz33** [theOehrly/Fast-F1#483] [Q&A]
Good evening everyone,

I updated my fastf1 to v3.1.5, however, i can't have access to Las Vegas (the frst under v3.1.5) but i still can download previous races.
I tried to delete the cache, same problem.
I tried with and without VPN

Does anyone have the same problem and know what can be the solution ?

Thank you !

Informations that can maybe useful :
I code on VSCode
Jupyter/Python 3.12.0
Windows 11

This is my typical code (wich work with Round 21 but not 22)  :

_from matplotlib import pyplot as plt
import fastf1
import fastf1.plotting
import numpy as np
import pandas as pd
from datetime import timedelta_

_session = fastf1.get_session(2023,22,3)
session
session.load()_

And this is my pip list :
_Package            Version
------------------ ------------
argcomplete        3.1.6
asttokens          2.4.1
attrs              23.1.0
cattrs             23.2.1
certifi            2023.11.17
charset-normalizer 3.3.2
click              8.1.7
colorama           0.4.6
comm               0.2.0
contourpy          1.2.0
cycler             0.12.1
debugpy            1.8.0
decorator          5.1.1
executing          2.0.1
fastf1             3.1.5
fonttools          4.44.3
idna               3.4
ipykernel          6.26.0
ipython            8.17.2
jedi               0.19.1
jupyter_client     8.6.0
jupyter_core       5.5.0
kiwisolver         1.4.5
matplotlib         3.8.2
matplotlib-inline  0.1.6
nest-asyncio       1.5.8
numpy              1.26.2
packaging          23.2
pandas             2.1.3
parso              0.8.3
Pillow             10.1.0
pip                23.3.1
pipx               1.2.1
platformdirs       4.0.0
prompt-toolkit     3.0.41
psutil             5.9.6
pure-eval          0.2.2
Pygments           2.17.1
pyparsing          3.1.1
python-dateutil    2.8.2
pytz               2023.3.post1
pywin32            306
pyzmq              25.1.1
rapidfuzz          3.5.2
requests           2.31.0
requests-cache     1.1.1
scipy              1.11.4
six                1.16.0
stack-data         0.6.3
thefuzz            0.20.0
timple             0.1.7
tornado            6.3.3
traitlets          5.13.0
tzdata             2023.3
url-normalize      1.4.3
urllib3            2.1.0
userpath           1.9.1
wcwidth            0.2.10
websockets         12.0_

**Comment by theOehrly** [comment]
Ok, right away I notice two things. You are loading round number 22 in the example, which is Abu Dhabi (the next race) and not Las Vegas. F1.com counts list Las Vegas as Round 22 because they are still counting the cancelled Italian Emilia-Romagna GP. FastF1 does not count that GP, as the event has not happend (this more or less follows what the FIA is doing and what the Ergast database is doing).
Therefore, the above code snippet fails because the data isn't available.

Loading the correct round works for me.

Second (but unrelated), FastF1 does not support Pandas version 2.1.0 or higher (yet). You have version 2.1.3 installed. 
Pip should notify you about that as well with a message like ``fastf1 3.1.5 requires pandas<2.1.0,>=1.2.4, but you have pandas 2.1.3 which is incompatible``
You likely will be able to load the data, but other functionality will not be working correctly. 
You can downgrade using ``pip install pandas==2.0.3``. But that version needs to be built from source because it was release before Python 3.12 and therefore, there are no pre-built wheels for it. For that, you likely need to install additional build dependencies. Or you could downgrade to Python 3.11 which might be easier.

**Reply by Maxigaz33** [reply]
I was thinking it was working because when I "print(session)" it was printing "2023 Season Round 22: Las Vegas Grand Prix - Practice 3".
But anyway, I tried with "session = fastf1.get_session(2023,'Sao Paulo','Q')" and it's working
But when i try with "session = fastf1.get_session(2023,'Las Vegas','Q')" it's not working.

For the pandas : thanks i fixed it !

**Reply by theOehrly** [reply]
That is pretty weird, ``print(session)`` should print the correct round number according to the numbering scheme that FastF1 uses.

Please run the following code and send me the full output. Either just paste it here or use a service like pastebin.com

```
import fastf1
fastf1.set_log_level('DEBUG')
session = fastf1.get_session(2023, 'Las Vegas', 'Q')
session.load()
```
